wconsteroids/wconsteroids/StatisticsCollector.cpp

```diff
@@ -27,24 +27,23 @@ void StatisticsCollector::analyzeBuffer(std::string inputBuffer) noexcept
 	fileStatistics.setToLineCount(lineCount);
 	countWordsAndWhiteSpace(inputBuffer);
 
-	std::string::iterator currentChar = startBuffer;
+	auto currentChar = startBuffer;
 	while (currentChar != endBuffer)
 	{
 		updateWidestLine(currentChar, endBuffer);
 	}
 }
 
-void StatisticsCollector::countWordsAndWhiteSpace(std::string& inputBuffer) noexcept
+void StatisticsCollector::countWordsAndWhiteSpace(std::string_view inputBuffer) noexcept
 {
 	std::size_t wordCount = 0;
 	std::size_t whiteSpaceCount = 0;
-	std::string::iterator currentChar = inputBuffer.begin();
-	std::string::iterator endOfInput = inputBuffer.end();
+	auto currentChar = inputBuffer.begin();
+	auto endOfInput = inputBuffer.end();
 	bool inWord = false;
-
 	for ( ; currentChar != endOfInput; )
 	{
-		while (isspace(*currentChar))
+            while (std::isspace(static_cast<unsigned char>(*currentChar)))
```

Comment:
It would be a little neater to have it like this:
```
while (currentChar != endOfInput && std::isspace(static_cast<unsigned_char>(*currentChar)))
```
and then eliminate the awkward `break` clause.

Reply:
Also why is the line above a `for` loop instead of `while`?

Reply:
I left the `for` loop as-is, because I was just fixing the signed-char bug and the missing namespace name.  I didn't look beyond the line itself, but that's a good target for the next round.

Reply:
I missed the same problem 10 lines later - definitely adding that to the patch.